core/tee/tee_rpmb_fs.c

```diff
@@ -1045,57 +1104,56 @@ static TEE_Result tee_rpmb_write_and_verify_key(void)
 }
 #endif
 
-/* This function must never return TEE_SUCCESS if rpmb_ctx == NULL */
-static TEE_Result tee_rpmb_init(void)
+static TEE_Result rpmb_set_dev_info(const struct rpmb_dev_info *dev_info)
 {
-	TEE_Result res = TEE_SUCCESS;
-	struct rpmb_dev_info dev_info = { };
 	uint32_t nblocks = 0;
 
-	if (rpmb_dead)
-		return TEE_ERROR_COMMUNICATION;
+	DMSG("RPMB: Syncing device information");
 
-	if (!rpmb_ctx) {
-		rpmb_ctx = calloc(1, sizeof(struct tee_rpmb_ctx));
-		if (!rpmb_ctx)
-			return TEE_ERROR_OUT_OF_MEMORY;
-		rpmb_ctx->dev_id = CFG_RPMB_FS_DEV_ID;
-		rpmb_ctx->shm_type = THREAD_SHM_TYPE_APPLICATION;
-	}
+	DMSG("RPMB: RPMB size is %"PRIu8"*128 KB", dev_info->rpmb_size_mult);
+	DMSG("RPMB: Reliable Write Sector Count is %"PRIu8,
+	     dev_info->rel_wr_sec_c);
+	DMSG("RPMB: CID");
+	DHEXDUMP(dev_info->cid, sizeof(dev_info->cid));
 
+	if (!dev_info->rpmb_size_mult)
+		return TEE_ERROR_GENERIC;
 
-	if (!rpmb_ctx->dev_info_synced) {
-		DMSG("RPMB: Syncing device information");
+	if (MUL_OVERFLOW(dev_info->rpmb_size_mult,
+			 RPMB_SIZE_SINGLE / RPMB_DATA_SIZE, &nblocks) ||
+	    SUB_OVERFLOW(nblocks, 1, &rpmb_ctx->max_blk_idx))
+		return TEE_ERROR_BAD_PARAMETERS;
 
-		dev_info.rpmb_size_mult = 0;
-		dev_info.rel_wr_sec_c = 0;
-		res = tee_rpmb_get_dev_info(&dev_info);
-		if (res != TEE_SUCCESS)
-			goto func_exit;
+	memcpy(rpmb_ctx->cid, dev_info->cid, RPMB_EMMC_CID_SIZE);
 
-		DMSG("RPMB: RPMB size is %d*128 KB", dev_info.rpmb_size_mult);
-		DMSG("RPMB: Reliable Write Sector Count is %d",
-		     dev_info.rel_wr_sec_c);
+	if (IS_ENABLED(RPMB_DRIVER_MULTIPLE_WRITE_FIXED))
```

Comment:
Perhaps this conditional could be removed? I can see it dates back to commit b01047730e77 ("Open-source the TEE Core") so it may very well be irrelevant now.

Reply:
I'd rather remove that in a separate commit, if it should be removed at all. I don't see any harm in keeping it though, it explains what `rel_wr_sec_c` is supposed to be used for even if we don't trust it at the moment.

Reply:
ok